core/capabilities/webapi/target/capability.go

```diff
@@ -0,0 +1,172 @@
+package target
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/validation"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/workflow"
+)
+
+var _ capabilities.TargetCapability = &Capability{}
+
+type Capability struct {
+	gc            connector.GatewayConnector
+	lggr          logger.Logger
+	registry      core.CapabilitiesRegistry
+	config        Config
+	responseChs   map[string]chan *api.Message
+	responseChsMu *sync.Mutex
+}
+
+func NewCapability(config Config, registry core.CapabilitiesRegistry, gc connector.GatewayConnector, lggr logger.Logger,
+	responseChs map[string]chan *api.Message, responseChsMu *sync.Mutex) (*Capability, error) {
+	return &Capability{
+		responseChs:   make(map[string]chan *api.Message),
+		responseChsMu: responseChsMu,
+	}, nil
+}
+
+func (c *Capability) Start(ctx context.Context) error {
+	return c.registry.Add(ctx, c)
+}
+
+func (c *Capability) Close() error {
+	return nil
+}
+
+func (c *Capability) Info(ctx context.Context) (capabilities.CapabilityInfo, error) {
+	return capabilities.CapabilityInfo{}, nil
+}
+
+func getMessageID(req capabilities.CapabilityRequest) (string, error) {
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowID); err != nil {
+		return "", fmt.Errorf("workflow ID is invalid: %w", err)
+	}
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowExecutionID); err != nil {
+		return "", fmt.Errorf("workflow execution ID is invalid: %w", err)
+	}
+	messageID := []string{
+		req.Metadata.WorkflowID,
+		req.Metadata.WorkflowExecutionID,
+		workflow.MethodWebAPITarget,
+	}
+	return strings.Join(messageID, "/"), nil
+}
+
+func (c *Capability) Execute(ctx context.Context, req capabilities.CapabilityRequest) (capabilities.CapabilityResponse, error) {
+	c.lggr.Debugw("executing http target", "capabilityRequest", req)
+
+	var input WorkflowInput
```

Comment:
Calling this WorkflowInput is slightly misleading IMO: the capability defines the input, not the workflow

Reply:
makes sense. this can be webapi.TargetInput if we start enforcing the convention above.